### PR TITLE
Fix sorting on brigades (map) page AND get user location

### LINKED ISSUE #151: Sorting on Brigades page
In the table next to the map.
Sorting is case sensitive.
And when I sort by brigade it seems to sort by project.
@rkaufman13 also says that sorting by project sort first by brigade and then by project.

### PR DESCRIPTION
**Pull request by rkaufman13**
Fixes #151
 
Sorting is now case-insensitive and defaults to sort by project.
Doing this, however, surfaced an odd annoyance — there are two deprecated projects by CodeNamu in Seoul that begin with a hyphen, so they appear first in the list, even if the Seoul pin isn't visible (maybe on larger screens than I have Seoul is visible). Anyway, the easiest fix was to add an ask for the user's location and center the pin there. (We can also discuss if we also want to zoom in on the user's location or if we still want to show mostly the whole U.S.)

As I'm thinking about it another option would be to hide old projects by default on the map page, as we do on the main projects page....

### CODE REVIEW

File: statusboard/src/components/Map/Map.js [hunk 1]
```diff
@@ -25,14 +27,32 @@ export default function Map({ brigadeData, filterOpts, setFilterOpts }) {
...
+        userCenter = [position.coords.latitude, position.coords.longitude];
+        setZoom(2);
+        setCenter(userCenter);
+      };
+      const noLocation = () => {
+        setZoom(defaultZoom);
+        setCenter(defaultCenter);
+      };
+
+      const options = { timeOut: 20000, maximumAge: 60 * 60 * 24 };
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline statusboard/src/components/Map/Map.js:41]: [Add 1000 to `maximumAge`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition#:~:text=maximum%20age%20in%20milliseconds)

**Review by k3KAW8Pnf7mkmdSMPHz27 — CHANGES_REQUESTED** [review]
It looks good and I like the features as well, and well below a dozen comments including nitpicks X)

File: statusboard/src/components/Map/Map.js [hunk 2]
```diff
@@ -25,14 +27,32 @@ export default function Map({ brigadeData, filterOpts, setFilterOpts }) {
       const { latitude, longitude } = filterOpts.selectedBrigade || {};
       setCenter([latitude, longitude]);
     } else {
-      setZoom(defaultZoom);
-      setCenter(defaultCenter);
+      let userCenter = [];
+      const foundLocation = (position) => {
+        userCenter = [position.coords.latitude, position.coords.longitude];
```
[the remark this answers is not in the document]
Comment (k3KAW8Pnf7mkmdSMPHz27) [reply statusboard/src/components/Map/Map.js:32]: I like the naming convention, it really helps reading the code

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
> Doing this, however, surfaced an odd annoyance — there are two deprecated projects by CodeNamu in Seoul that begin with a hyphen, so they appear first in the list, even if the Seoul pin isn't visible (maybe on larger screens than I have Seoul is visible). Anyway, the easiest fix was to add an ask for the user's location and center the pin there. (We can also discuss if we also want to zoom in on the user's location or if we still want to show mostly the whole U.S.)

What is your opinion on zooming to the user's location?

> As I'm thinking about it another option would be to hide old projects by default on the map page, as we do on the main projects page....

I'd consider that the easiest fix that can be implemented fairly quickly.

Long term I was thinking of doing something similar to [alltrails](https://www.alltrails.com/explore), but I probably should try to read through https://github.com/alltrails/AllTrailsLunch-1 to know a bit more about what is needed :stuck_out_tongue:

File: statusboard/src/utils/types.ts [hunk 3]
```diff
@@ -41,3 +45,6 @@ export type Location = {
     longitude?: string | number;
   };
 };
+
+
+
```
[the remark this answers is not in the document]
Comment (rkaufman13) [reply statusboard/src/utils/types.ts]: Oops, I don't know why Prettier auto-format-on-save keeps turning itself off.

File: statusboard/src/components/Map/Map.js [hunk 4]
```diff
@@ -25,14 +27,32 @@ export default function Map({ brigadeData, filterOpts, setFilterOpts }) {
...
+        options
+      );
     }
+
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [selectedBrigadeName]);
 
   return (
     <div className="map leaflet-container">
+      {}
```
[the remark this answers is not in the document]
Comment (rkaufman13) [reply statusboard/src/components/Map/Map.js]: Oops, great catch--I was going to conditionally render the map and then decided to use an effect hook instead. Removed

File: statusboard/src/components/Map/Map.js (hunk 1 above)
Comment (rkaufman13) [reply statusboard/src/components/Map/Map.js]: Yes 🙃

Comment (rkaufman13) [reply statusboard/src/components/Map/Map.js]: (Fixed)

File: statusboard/src/utils/types.ts [hunk 5]
```diff
@@ -1,4 +1,8 @@
 // If we also use typescript on the back end, we can move this into a "common" folder (still within the front end bc CRA is opinionated) and use it there too
+import { SortByFn,
+  Row,
+  IdType, } from "react-table";
+import { customStringSort,lastPushSort } from "./utils";
```
[the remark this answers is not in the document]
Comment (rkaufman13) [reply statusboard/src/utils/types.ts]: > What is your opinion on zooming to the user's location?

This is a great question. I kind of like seeing the whole U.S. map when I go to the page. It gives a great sense of the scale of all the brigades across the country, not to mention worldwide civic tech groups.

That said, I'm not sure that, for example, West Coast folks would see the whole map at this zoom if they provide their location. I don't know **anything** about spoofing my location so I'm not able to test how it looks. Do you know how to do it? If it looks bad for half the country I could see an argument for zooming in if location is provided.

> > As I'm thinking about it another option would be to hide old projects by default on the map page, as we do on the main projects page....
> 
> I'd consider that the easiest fix that can be implemented fairly quickly.
> 
> Long term I was thinking of doing something similar to [alltrails](https://www.alltrails.com/explore), but I probably should try to read through https://github.com/alltrails/AllTrailsLunch-1 to know a bit more about what is needed 😛

I can make a separate PR for adding a "in the last week/month/year" dropdown and defaulting to last year. I think it won't be too involved but it's definitely a separate issue as `brigades` currently has no filtering on it whatsoever.

File: statusboard/src/components/Map/Map.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rkaufman13) [reply statusboard/src/components/Map/Map.js:32]: Sure thought I could, based on my understanding of how arrays work in JS! However, the linter complained and failed to compile so we're back at let for now.

File: statusboard/src/utils/types.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (k3KAW8Pnf7mkmdSMPHz27) [reply statusboard/src/utils/types.ts]: > ...  spoofing my location so I'm not able to test how it looks. Do you know how to do it?

To be frank, I'd hardcode it and test it by using different hard-coded values :flushed: 
The "correct" way is likely something like this if your in Chrome https://developer.chrome.com/docs/devtools/device-mode/geolocation/

> I can make a separate PR for adding ...

That sounds good. I'll probably think it is justified with a Brigades/CfA only filter in the future. Perhaps it is worth reversing the default value though (i.e., show all projects by default)

File: statusboard/src/components/Map/Map.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (k3KAW8Pnf7mkmdSMPHz27) [reply statusboard/src/components/Map/Map.js:32]: That is weird. Copying the change (i.e., removing line 30 completely and just adding `const` in front of `userCenter`) compiles for me locally. What error is it that you are getting?

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
Looks good to me, and thank you for contributing both code and ideas! :grinning:

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
Since the `let -> const` was a nitpick and we know what was going on I'll merge it now!

**Pull request merged**